Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -709,7 +705,7 @@ public ManaCosts<ManaCost> getManaCosts() {
      */
     @Override
     public ManaCosts<ManaCost> getManaCostsToPay() {
-        return manaCostsToPay;
+        return manaCostsToPay != null ? manaCostsToPay : emptyManaCosts;
```

Comment:
Don't use that style of code (null init and return empty object on null). It's gives much more hidden and hard find bugs (e.g. card's code can change a manacost, but nothing happen due nullable or empty value). Costs are popular fields, so it must be init from the start (more easy and readable code without null-protection). It's has zero effect on performance.

Also try not to use changeable objects as constant -- it's potentially very buggy. Example: const object modification bug for all server:
![shot_221007_182325](https://user-images.githubusercontent.com/8344157/194576713-3a652675-4171-4e48-aa94-38c13f1dba46.png)

Reply:
Yes, the modifiable default is a big problem. I rolled that part back (see comment in main thread). Some of the other fields that I replaced return a non-modifiable default value